DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Models;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class RegisterController : Controller
+    {
+        private const string CookieName = "RegistrationData";
+        private readonly ICentresService centresService;
+        private readonly IJobGroupsService jobGroupsService;
+
+        public RegisterController(ICentresService centresService, IJobGroupsService jobGroupsService)
+        {
+            this.centresService = centresService;
+            this.jobGroupsService = jobGroupsService;
+        }
+
+        public IActionResult Index()
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            var delegateRegistrationData = TempData.Get<DelegateRegistrationData>();
+
+            if (delegateRegistrationData == null || !Request.Cookies.ContainsKey(CookieName))
+            {
+                var id = Guid.NewGuid();
+                delegateRegistrationData = new DelegateRegistrationData { Id = id };
+
+                Response.Cookies.Append(
+                    CookieName,
+                    id.ToString(),
+                    new CookieOptions
+                    {
+                        Expires = DateTimeOffset.UtcNow.AddDays(30)

```

Comment:
Any reason for the choice of 30? Also, should we pull the 30 into variable at the top of the file (e.g. registrationCookieExpiryTimeInDays) for extra emphasis?

Reply:
I'm not totally sure why we need a cookie anyway. [The example multi-page form](https://github.com/tomdoughty/nhsuk.base-application/blob/main/src/nhsuk.base-application/Controllers/ExampleFormController.cs) sets one for 30 days but it looks like TempData is stored in a cookie anyway, and we never use this id for anything, so I don't know if we need it.

Reply:
Ah, interesting! Do you think there'll be some issues storing the password data in a cookie (especially given the non-hashed form)?

Reply:
Documentation is really sparse as to what TempData is and how it works but it _looks_ like the cookie is encrypted, so I don't think this will be a problem, especially if we make sure it's cleared once they actually register.

Reply:
I've finally found [some concrete docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/app-state?view=aspnetcore-3.1#tempdata-providers) on it, and TempData cookies are encrypted! I'll make sure it's only ever stored hashed as part of the backend work and then I think everything will be fine.

Reply:
Could we add a TODO for hashing the password just in case? 🙏

Reply:
Added a TODO for this